extensions/notebook/src/dialog/addNotebookDialog.ts

```diff
@@ -0,0 +1,179 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as path from 'path';
+import { pathExists, remove } from 'fs-extra';
+import * as loc from '../common/localizedConstants';
+import { IconPathHelper } from '../common/iconHelper';
+import { FileType, IBookTocManager } from '../book/bookTocManager';
+import { confirmReplace, getDropdownValue } from '../common/utils';
+import { IPrompter } from '../prompts/question';
+import CodeAdapter from '../prompts/adapter';
+import { BookModel } from '../book/bookModel';
+import { BookTreeItemType } from '../book/bookTreeItem';
+
+export class AddNotebookDialog {
+	private dialog: azdata.window.Dialog;
+	public view: azdata.ModelView;
+	private formModel: azdata.FormContainer;
+	private notebookNameInputBox: azdata.InputBoxComponent;
+	private booksDropdown: azdata.DropDownComponent;
+	private notebookTypeDropdown: azdata.DropDownComponent;
+	private saveLocationInputBox: azdata.InputBoxComponent;
+	private prompter: IPrompter;
+	private bookNames: string[];
+
+	constructor(private tocManager: IBookTocManager, private books: BookModel[]) {
+		this.prompter = new CodeAdapter();
+		this.bookNames = this.books.length > 0 ? this.books.map(b => b.bookItems[0].title) : [];
```

Comment:
You don't have to use the ternary - if this.books is empty it'll just return any empty array anyways